### PR TITLE
Fix floating point argument conversion in bun:ffi

### PR DESCRIPTION
**Pull request by jjrv**
### What does this PR do?

Don't try to convert JS values to floating point when calling a function through FFI that expects a double. Instead, handle integer and floating point values in the generated C wrapper code.

See #8430

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

- [X] I included a test for the new code, or existing tests cover it
- [X] I ran my tests locally and they pass (`bun-debug test test-file-name.test`)

### CODE REVIEW

**Comment by jjrv** [comment]
Still need to figure out why the other two tests are broken on my machine. However they're already broken before the code change in this PR. Based on the required `Makefile` changes, the tests may not have been run on Linux before?

**Comment by jjrv** [comment]
Still need to figure out why the other two tests are broken on my machine. However they're already broken before the code change in this PR. Based on the required `Makefile` changes, the tests may not have been run on Linux before?

**Comment by jjrv** [comment]
Still need to figure out why the other two tests are broken on my machine. However they're already broken before the code change in this PR. Based on the required `Makefile` changes, the tests may not have been run on Linux before?

File: test/js/bun/ffi/ffi.test.js [hunk 1]
```diff
@@ -599,7 +601,7 @@ function ffiRunner(fast) {
           expect(bigint ? BigInt(fn(0n)) : fn(0)).toBe(bigint ? 0n : 0);
 
           for (let i = min; i <= max; i += inc) {
-            expect(bigint ? BigInt(fn(i)) : fn(i)).toBe(i);
+            // expect(bigint ? BigInt(fn(i)) : fn(i)).toBe(i);
```
Comment (jjrv) [inline test/js/bun/ffi/ffi.test.js]: This was oddly already failing:

```
Expected: 2147483648
Received: -2147483648

run ffi > FFI runner (fast int) > integer identities work for all possible values > uint32_t
```

**Comment by paperclover** [comment]
one thing jarred mentioned to me about about the approach of doing the check in C is 'how does it impact performance?'

could you do a benchmark in a release build?

File: test/js/bun/ffi/ffi.test.js (hunk 1 above)
Comment (paperclover) [reply test/js/bun/ffi/ffi.test.js]: this was already broken. i had a pr to fix a different issue with uint32 but it causes another failure. which is funny because this is the test i wrote that is now failing because of that pr.

i do a stupid trick for uint32 and the sign bit is the edge case

**Comment by jjrv** [comment]
> one thing jarred mentioned to me about about the approach of doing the check in C is 'how does it impact performance?'

I added a branch in autogenerated C code and removed a branch from the JS function calling it. It shouldn't make it worse. Is there an existing benchmark or is there reasonable doubt that branches in C are worse?

Also, I *can* make the C branchless using bit twiddling hacks, but doubt it's worth it. That *would* need benchmarking.

File: test/js/bun/ffi/ffi.test.js [hunk 2]
```diff
@@ -605,6 +607,27 @@ function ffiRunner(fast) {
       }
     });
 
+    describe("Floating point identities for all valid exponents", () => {
+      const cases = [
+        // { type: "float", min: -149, max: 128, fn: identity_float },
```
Comment (jjrv) [inline test/js/bun/ffi/ffi.test.js:612]: Okayyy, it also corrupts floats so this test fails.

**Comment by jjrv** [comment]
Hmm, with the latest test it just crashes when there's too many `expect` calls :/

File: test/js/bun/ffi/ffi.test.js (hunk 1 above)
Comment (jjrv) [reply test/js/bun/ffi/ffi.test.js]: I think I fixed it now, but the whole test has become unstable and I suspect it's not because of the number conversions but because it's making too many calls to either FFI or `expect` and something goes wrong

**Comment by jjrv** [comment]
Well, I'm giving up for now due to lack of time. Can't get the tests to run, even though I'm pretty sure the double handling would have been improved. GDB reports:

```
✓ run ffi > FFI runner (fast int) > integer identities work for all possible values > uint64_t [568.95ms]
[jest] test("primitives")

Thread 1 "bun-debug" received signal SIGSEGV, Segmentation fault.
0x0000020000905953 in ?? ()
(gdb) bt
#0  0x0000020000905953 in ?? ()
#1  0x00007fffad408038 in ?? ()
#2  0x00007ffffffe4c30 in ?? ()
#3  0x000055555f005876 in js_trampoline_op_call ()
#4  0x0000000000000000 in ?? ()
```
Looks like an FFI call jumps to a null pointer? Since it always happens after a specific number of tests, maybe something lived in a buffer that got reclaimed? Anyway it'd take some work to find the cause.
